### PR TITLE
Build and dependency changes

### PR DESCRIPTION
**Pull request by harissecic**
Cleaned up build and merged with new changes remove Pulsar library dependencies on netty, jackson, and gson and allowing it to use Micronaut ones.

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -8,6 +8,9 @@ spockVersion=2.0-groovy-3.0
 kotlinVersion=1.6.20
 kotlinCoroutinesVersion=1.6.1
 micronautBuildVersion=1.1.5
+micronautGradlePluginVersion=2.0.3
```
Comment (melix) [inline gradle.properties:11]: :x: those should be handled in the libs.versions.toml file

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -5,13 +5,13 @@ kotest = '4.6.4'
...
 micronaut-serde = '1.0.1'
+micronaut-docs = '2.0.0'
 micronaut-gradle = '3.4.1'
-
 managed-pulsar = '2.10.0'
 
 [libraries]
 # Managed libraries
-managed-pulsar = { module = 'org.apache.pulsar:pulsar-client', version.ref= 'managed-pulsar' }
+managed-pulsar = { module = 'org.apache.pulsar:pulsar-client-original', version.ref= 'managed-pulsar' }
```
Comment (melix) [inline gradle/libs.versions.toml:14]: Maybe we should change the alias name to `managed-pulsar-client` to make it clearer.

File: buildSrc/src/main/groovy/io.micronaut.build.internal.pulsar-module.gradle [hunk 3]
```diff
@@ -7,6 +7,19 @@ repositories {
...
 dependencies {
+    api mn.micronaut.messaging
+    api libs.micronaut.serde.support
+    api(libs.managed.pulsar) {
+        exclude group: 'io.netty'
+        exclude group: 'com.fasterxml.jackson.module'
+        exclude group: 'com.google.code.gson'
+    }
+
+    implementation(libs.managed.pulsar) {
```
Comment (melix) [inline buildSrc/src/main/groovy/io.micronaut.build.internal.pulsar-module.gradle:18]: :x: it doesn't make sense to have the same dependency in both `api` and `implementation`: it should only be in `api`.

**Review by melix — CHANGES_REQUESTED** [review]
Thanks for the update, I only took a look at the build aspects, I cannot really judge the other changes.

**Comment by harissecic** [comment]
> Thanks for the update, I only took a look at the build aspects, I cannot really judge the other changes.

I've updated those, as for the rest of the changes I was mainly worried how will `EventLoopGroup` be injected in case someone uses `platform("tomcat")` with the application. Not sure how is netty dependency resolved in the Micronaut core if we use different platform.

File: gradle/updates/libs.versions.toml [hunk 4]
```diff
@@ -15,7 +15,7 @@ kotlin-stdlib = { module = 'org.jetbrains.kotlin:kotlin-stdlib-jdk8', version.re
 testcontainers = { module = 'org.testcontainers:testcontainers', version.ref = 'testcontainers' }
 testcontainers-spock = { module = 'org.testcontainers:spock', version.ref = 'testcontainers' }
 testcontainers-pulsar = { module = 'org.testcontainers:pulsar', version.ref = 'testcontainers' }
-pulsar = { module = 'org.apache.pulsar:pulsar-client-original', version.ref= 'pulsar' }
+managed-pulsar-client = { module = 'org.apache.pulsar:pulsar-client-original', version.ref= 'pulsar' }
```
Comment (melix) [inline gradle/updates/libs.versions.toml:18]: the version ref should be `managed-pulsar-client`.

**Pull request merged**